lib/logger.js

```diff
@@ -13,17 +13,34 @@ var DEF_LEVEL_INDEX = LEVELS.indexOf(DEF_LEVEL);
 
 // Simple bunyan logger wrapper
 function Logger(confOrLogger, args) {
+    var self = this;
     if (confOrLogger.constructor !== Logger) {
         // Create a new root logger
         var conf = this._processConf(confOrLogger);
-        this._logger = bunyan.createLogger(conf);
-        this._levelMatcher = this._levelToMatcher(conf.level);
+        self._log_components = conf.log_components || {};
+        delete conf.log_components;
```

Comment:
The original config is not modified, as it's copied wishing `_processConf` method.